mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -167,51 +141,39 @@ void SpeedometerData::LoadGamemodeData(GameMode_t gametype)
     }
 }
 
-void SpeedometerData::SaveGamemodeData() 
-{ 
-    SaveGamemodeData(g_pGameModeSystem->GetGameMode()->GetType());
+void SpeedometerData::SetVisible(GameMode_t gametype, SpeedometerLabel_t speedometerLabelType, bool bVisible) 
+{
+    GetSpeedoKVs(gametype, speedometerLabelType)->SetBool(SPEEDOMETER_KEY_VISIBLE, bVisible);
 }
 
-void SpeedometerData::SaveGamemodeData(int gametype)
+bool SpeedometerData::GetVisible(GameMode_t gametype, SpeedometerLabel_t speedometerLabelType) const
 {
-    if (gametype >= 0 && gametype < GAMEMODE_COUNT)
-        SaveGamemodeData(GameMode_t(gametype));
+    return GetSpeedoKVs(gametype, speedometerLabelType)->GetBool(SPEEDOMETER_KEY_VISIBLE, true);
 }
 
-void SpeedometerData::SaveGamemodeData(GameMode_t gametype)
+void SpeedometerData::SetColorize(GameMode_t gametype, SpeedometerLabel_t speedometerLabelType, SpeedometerColorize_t speedometerColorizeType)
 {
-    const auto pGamemodeKV = m_pGamemodeSetupData->FindKey(g_szGameModes[gametype], true);
-
-    // set speedometer label setups
-    FOR_EACH_SUBKEY(pGamemodeKV, pSpeedoItem)
-    {
-        const char *speedoname = pSpeedoItem->GetName();
-        SpeedometerLabel *pLabel = GetLabelFromName(speedoname);
-        KeyValues *pSpeedoKV = pGamemodeKV->FindKey(speedoname, true);
-        if (!pLabel || !pSpeedoKV)
-            continue;
-
-        pLabel->SaveToKV(pSpeedoKV);
-    }
+    GetSpeedoKVs(gametype, speedometerLabelType)->SetInt(SPEEDOMETER_KEY_COLORIZE, speedometerColorizeType);
+}
 
-    // set ordering
-    auto pOrderList = g_pSpeedometer->GetLabelOrderListPtr();
-    KeyValues *pOrderKVs = pGamemodeKV->FindKey("order", true);
-    pOrderKVs->Clear();
-    for (auto i = 0; i < pOrderList->Count(); i++)
-    {
-        pOrderKVs->SetString(CFmtStr("%i", i + 1).Get(), (*pOrderList)[i]->GetName());
-    }
+SpeedometerColorize_t SpeedometerData::GetColorize(GameMode_t gametype, SpeedometerLabel_t speedometerLabelType) const
+{
+    return SpeedometerColorize_t(GetSpeedoKVs(gametype, speedometerLabelType)->GetInt("colorize", SPEEDOMETER_COLORIZE_COMPARISON));
```

Comment:
Looks like you forgot to use the macro here

Reply:
big oops